src/PaymentGateways/Gateways/Stripe/LegacyDonationData.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe;
+
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+
+/**
+ * @unreleased
+ */
+class LegacyDonationData
```

Comment:
Let's move these Legacy prefixed classes into the `LegacyPaymentGateways` domain.  That's where our legacy adapters and purgatory items live that kind of bridge the gap into our new api.  Let's try to keep our PaymentGateways domain as clean as possible.

Reply:
@jonwaldstein I was _almost_ able to refactor out the need for the `LegacyDonationData` class, but alas the format is needed for backwards compatibility with the filter used when preparing Stripe metadata.

Reply:
That said, I have relocated the `LegacyDonationData` to `src/LegacyPaymentGateways/Adapters`.